### PR TITLE
Pre Quote Design Iteration

### PR DESCRIPTION
**Pull request by nalancer08**
### Type of change

- [ ] Chore
- [X] Feature
- [ ] Bug fix

### Linked issue

- [X] Not tracked

### Why

Faramarz new design iteration

Includes:

- New desing of Faramarz
- Michael's comments over the new design

### Note

The icons are hosted locally in the app, 'cuase we dont have server hosted png images for crypto images!!!!

### Evidence

imagen

### CODE REVIEW

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt [hunk 1]
```diff
@@ -108,317 +102,442 @@ class TradeFlow @JvmOverloads constructor(
...
+    private fun CryptoCurrencyInput(
+        currencyState: MutableState<AssetBankModel>,
+        expandedCurrencyInput: MutableState<Boolean>,
+        currencyInputWidth: MutableState<Size>) {
+
+        Text(
+            modifier = Modifier
+                .padding(top = 27.dp)
+                .padding(horizontal = 1.dp),
+            text = "Crypto Currency",
```
Comment (brent-cybrid) [inline SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt:217]: Text always needs to be localized

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt [hunk 2]
```diff
@@ -108,317 +102,442 @@ class TradeFlow @JvmOverloads constructor(
...
+                    val expandedCurrencyInput = remember { mutableStateOf(false) }
+                    val currencyInputWidth = remember { mutableStateOf(Size.Zero) }
+
+                    // -- Value Input
+                    val typeOfValueState = remember { mutableStateOf(AssetBankModel.Type.fiat) }
+                    val valueState = remember { mutableStateOf("") }
+                    val valueAsset = remember { mutableStateOf(
+                        if (typeOfValueState.value == AssetBankModel.Type.fiat)
+                            pairAsset else currencyState.value
+                    )}
```
Comment (brent-cybrid) [inline SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt:122]: What's going on here?

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt [hunk 3]
```diff
@@ -108,317 +102,442 @@ class TradeFlow @JvmOverloads constructor(
...
+
+                    CryptoHeaderTabs(
```
Comment (brent-cybrid) [inline SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt:124]: Why do all these classes have the prefix `Crypto`?

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt [hunk 4]
```diff
@@ -108,317 +102,442 @@ class TradeFlow @JvmOverloads constructor(
...
+                    )
+
+                    CryptoCurrencyDropDown(
+                        currencyState = currencyState,
+                        expandedCurrencyInput = expandedCurrencyInput,
+                        currencyInputWidth = currencyInputWidth,
+                        cryptoList = cryptoList
+                    )
+
+                    if (!expandedCurrencyInput.value) {
```
Comment (brent-cybrid) [inline SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt:142]: What's meant by `expandedCurrencyInput`?

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt [hunk 5]
```diff
@@ -108,317 +102,442 @@ class TradeFlow @JvmOverloads constructor(
...
-                                    }
-                                }
-                            },
-                        label = {
-                            Text(
-                                text = "Currency"
-                            )
-                        },
+    @Composable
+    private fun CryptoCurrencyInput(
```
Comment (brent-cybrid) [inline SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt:208]: We need to support the input of either the asset or the counterasset. We should stay away from terms like crypto and fiat if we can and stick with asset and counter asset. It's more generic and will avoid special casing.

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt [hunk 6]
```diff
@@ -108,317 +102,442 @@ class TradeFlow @JvmOverloads constructor(
...
-                                )
-                                val value = baseValue.divL(buyPriceDecimal)
-                                amount = value.toPlainString()
-                            }
+            AssetBankModel.Type.crypto -> {
 
-                            else -> {}
-                        }
+                valueAsset.value = currencyState.value
+                codeAssetToUse = pairAsset
```
Comment (brent-cybrid) [inline SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt:438]: What's `valueAsset` and `codeAsset`?

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt [hunk 7]
```diff
@@ -16,9 +12,7 @@ import androidx.compose.material.*
 import androidx.compose.material.TabRowDefaults.tabIndicatorOffset
 import androidx.compose.material.icons.Icons
 import androidx.compose.material.icons.filled.ArrowDropDown
-import androidx.compose.runtime.LaunchedEffect
-import androidx.compose.runtime.mutableStateOf
-import androidx.compose.runtime.remember
+import androidx.compose.runtime.*
```
[the remark this answers is not in the document]
Comment (nalancer08) [reply SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt:15]: Android Studio Linter add this "*", casue we literally are using all the imports

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt (hunk 1 above)
Comment (nalancer08) [reply SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt:217]: Solved!

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt (hunk 2 above)
Comment (nalancer08) [reply SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt:122]: We have the MutableState of AssentBankModel, but to assign the first value, we have this if that check the "typeOfValueState" (check if the input its accepting fiat or crypto) and depend of this to assign this model to get "code" and "name" for lables and placeholders

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt (hunk 3 above)
Comment (nalancer08) [reply SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt:124]: Chenged to PreQuote :)

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt (hunk 4 above)
Comment (nalancer08) [reply SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt:142]: This val is a:

`val expandedCurrencyInput = remember { mutableStateOf(false) }`

It means that handle if the curreny dropdownmenu its open or not

Captura de Pantalla 2022-06-17 a la(s) 13 33 49

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt (hunk 5 above)
Comment (nalancer08) [reply SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt:208]: Chenged to: PreQuoteCurrencyInput

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt (hunk 6 above)
Comment (nalancer08) [reply SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt:438]: valueAsset its used to handle the label text in the ValueInput to get this kind of text: "Amount ETH" or "Amount BTC"

codeAsset its useed to set the currency code besie the operation string result

Comment (brent-cybrid) [reply SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt:446]: Can we then use more descriptive variable names like `valueAssetLabel`?

Comment (nalancer08) [reply SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt:446]: Sure, but its not a label, valueAsset its an AssetBankModel
then, i created the label and i created like this:


```
    val valueHint = buildAnnotatedString {
    append(stringResource(id = R.string.trade_flow_text_field_amount_placeholder))
    withStyle(style = SpanStyle(
        color = colorResource(id = R.color.list_prices_asset_component_code_color))) {
        append(" (${valueAsset.value.code})")
    }
}
```

If you see im using tha valueAsset as mutable (value member) and the the code of the model

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt [hunk 8]
```diff
@@ -68,8 +67,8 @@ class TradeFlow @JvmOverloads constructor(
 
     init {
 
-        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
-        inflater.inflate(R.layout.trade_flow, this, true)
+        //val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
```
Comment (jc-cybrid) [inline SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt:70]: Avoid commented code.

**Review by jc-cybrid — COMMENTED** [review]
Just one Nit comment about commented code.

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt (hunk 8 above)
Comment (nalancer08) [reply SDKAndroid/src/main/java/app/cybrid/sdkandroid/flow/TradeFlow.kt:70]: Done!!

**Review by jc-cybrid — APPROVED** [review]

**Pull request merged**